xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/internal/document/DefaultDocumentAuthors.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.model.internal.document;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.xwiki.model.document.DocumentAuthors;
+import org.xwiki.user.UserReference;
+
+/**
+ * Default implementation of {@link DocumentAuthors}.
+ *
+ * @version $Id$
+ * @since 14.0RC1
+ */
+public class DefaultDocumentAuthors implements DocumentAuthors
+{
+    private UserReference contentAuthor;
+    private UserReference effectiveMetadataAuthor;
+    private UserReference originalMetadataAuthor;
+    private UserReference creator;
+
+    /**
+     * Default empty constructor.
+     */
+    public DefaultDocumentAuthors()
+    {
+    }
+
+    /**
+     * Default constructor cloning the given document authors.
+     *
+     * @param documentAuthors the authors information to be cloned.
+     */
+    public DefaultDocumentAuthors(DocumentAuthors documentAuthors)
+    {
+        if (documentAuthors != null) {
+            this.contentAuthor = documentAuthors.getContentAuthor();
+            this.effectiveMetadataAuthor = documentAuthors.getEffectiveMetadataAuthor();
+            this.originalMetadataAuthor = documentAuthors.getOriginalMetadataAuthor();
+            this.creator = documentAuthors.getCreator();
+        }
+    }
+
+    @Override
+    public void setContentAuthor(UserReference contentAuthor)
+    {
+        this.contentAuthor = contentAuthor;
```

Comment:
One thing which is missing from all the set* methods is to update the XWikiDocument metadataDirty flag.

Reply:
> One thing which is missing from all the set* methods is to update the XWikiDocument metadataDirty flag.

Yeah I noticed it, and was actually wondering about that. A way for dealing with that would be to introduce a dirty flag in `DefaultDocumentAuthors` and checking for its state when getting the flag value in `XWikiDocument` but wasn't sure if it was needed, or if we could consider that it would be the API user job to also set the flag when using the `DocumentAuthors` setters (in which case I should update the javadoc to indicate it). wdyt?

Reply:
> we could consider that it would be the API user job to also set the flag when using the DocumentAuthors setters (in which case I should update the javadoc to indicate it).

-1, nobody will remember that

What we do usually is embedding the XWikiDocument instance in the modificable object and call set dirty flag through it. See XWikiAttachment#setMetaDataDirty for example.